shotover-proxy/src/frame/cassandra.rs

```diff
@@ -545,3 +900,63 @@ pub struct CassandraBatch {
     serial_consistency: Option<Consistency>,
     timestamp: Option<CLong>,
 }
+
+#[cfg(test)]
+mod test {
+    use crate::frame::CQL;
+    use cql3_parser::cassandra_statement::CassandraStatement;
+
+    #[test]
+    fn cql_round_trip_test() {
+        let query = r#"BEGIN BATCH
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (2, 12, 'bar');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz');
+            APPLY BATCH;"#;
+
+        let expected = "BEGIN BATCH INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo'); INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (2, 12, 'bar'); INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz'); APPLY BATCH";
+        let cql = CQL::parse_from_string(query);
+        let result = cql.to_query_string();
+        assert_eq!(expected, result)
+    }
+
+    #[test]
+    fn cql_parse_multiple_test() {
+        let query = r#"INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (2, 12, 'bar');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz');"#;
+
+        let cql = CQL::parse_from_string(query);
+        assert_eq!(3, cql.statements.len());
+        for stmt in cql.statements {
+            if let CassandraStatement::Insert(_x) = stmt.statement {
+                // do nothing
+            } else {
+                panic!(
+                    "{:?}  should have been CassandraStatement::Insert",
+                    stmt.statement
+                );
+            }
+        }
+    }
+
+    #[test]
+    fn cql_bad_statement_test() {
+        let query = r#"INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name)  (2, 12, 'bar');
+                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz');"#;
+
+        let cql = CQL::parse_from_string(query);
+        assert_eq!(3, cql.statements.len());
+        for stmt in cql.statements {
+            if let CassandraStatement::Unknown(_x) = stmt.statement {
+                // do nothing
+            } else {
+                panic!(
+                    "{:?}  should have been CassandraStatement::Unknown",
+                    stmt.statement
+                );
+            }
+        }
```

Comment:
This should be implemented like:
```rust
    #[test]
    fn cql_bad_statement_test() {
        let query = r#"INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo');
                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name)  (2, 12, 'bar');
                INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz');"#;

        let cql = CQL::parse_from_string(query);
        assert_eq!(
            vec!(
                CQLStatement { statement: CassandraStatement::Unknown("INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (1, 11, 'foo')".to_string()) },
                CQLStatement { statement: CassandraStatement::Unknown("INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (2, 12, 'bar')".to_string()) },
                CQLStatement { statement: CassandraStatement::Unknown("INSERT INTO test_cache_keyspace_batch_insert.test_table (id, x, name) VALUES (3, 13, 'baz')".to_string()) },
            ),
            cql.statements
        );
    }
```

So that we actually assert on the contents of the Unknown.
You can also see that the test now fails, I think due to the problem resolved by https://github.com/shotover/shotover-proxy/pull/598/files#r869773919